internal/controller/finrestore_controller.go

```diff
@@ -33,16 +77,143 @@ type FinRestoreReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.17.3/pkg/reconcile
 func (r *FinRestoreReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	var restore finv1.FinRestore
+	err := r.Get(ctx, req.NamespacedName, &restore)
+	if k8serrs.IsNotFound(err) {
+		logger.Info("FinRestore resource not found", "error", err)
+		return ctrl.Result{}, nil
+	}
+	if err != nil {
+		logger.Error(err, "failed to get FinRestore")
+		return ctrl.Result{}, err
+	}
+
+	if restore.GetNamespace() != r.cephClusterNamespace {
+		logger.Info("restore is not managed by the target Ceph cluster",
+			"restore", restore.Name, "namespace", restore.Namespace, "clusterNamespace", r.cephClusterNamespace)
+		return ctrl.Result{}, nil
+	}
+
+	if restore.DeletionTimestamp.IsZero() {
+		return r.restore(ctx, &restore)
+	} else {
+		return r.cleanup(ctx, &restore)
+	}
+}
+
+func (r *FinRestoreReconciler) restore(ctx context.Context, restore *finv1.FinRestore) (ctrl.Result, error) {
```

Comment:
I would like to clarify that this method is a "reconcile method." I think that the reconciler does not directly restore anything.

```suggestion
func (r *FinRestoreReconciler) reconcileRestore(ctx context.Context, restore *finv1.FinRestore) (ctrl.Result, error) {
```